### ISSUE TITLE
Parsing doesn't support trailing commas

### ISSUE THREAD

**Issue by Kixiron** [vmware/differential-datalog#735]
There's a few things I've found that don't accept trailing commas that should:

Structure definitions:
```prolog
typedef Thing = Thing {
    field: type,
    field2: type2,
}
```
Structure creation:
```prolog
insert Thing(
    .field = 10,
    .field2 = 20,
);
```
Matches:
```prolog
match (something) {
    A -> 10,
    B -> 20,
}
```
Destructuring:
```prolog
A(b) :- C(D { e, f, })
```
Relations:
```prolog
relation A(b: C, d: E,)
```

While some of these examples look ugly because they're one-liners, it can be nice to have trailing commas with longer things that spill over lines

**Comment by ryzhyk** [comment]
Yes, trailing commas are currently not allowed.  I think, they are a fairly unique feature in Rust, and are even enforced by `rustfmt`.  Not sure how many other languages support them.  I initially found them ugly, but having used them for a while I tend to agree they are a good idea and we should probably support them.

**Comment by ryzhyk** [comment]
Addressed by #789
